src/liquidity_mining/liquidity_mining.rs

```diff
@@ -133,11 +136,20 @@ pub fn calculate_global_farm_rewards<Period: num_traits::CheckedSub + TryInto<u3
         .checked_mul_int(total_shares_z)
         .ok_or(MathError::Overflow)?;
 
-    let reward_per_period = yield_per_period
-        .checked_mul(&FixedU128::from(total_shares_z_adjusted))
+    let periods = TryInto::<u128>::try_into(periods_since_last_update).map_err(|_e| MathError::Overflow)?;
+
+    //(total_shares_z_adjusted * yield_per_period.into_inner() * periods)/FixedU128::DIV;
+    let calculated_rewards = to_u256!(total_shares_z_adjusted)
+        .checked_mul(yield_per_period.into_inner().into())
         .ok_or(MathError::Overflow)?
```

Comment:
niptick, you can use `U128::full_mul` here, so
```suggestion
    let calculated_rewards = U128::from(total_shares_z_adjusted)
        .full_mul(yield_per_period.into_inner().into())
```